### PR TITLE
Homework5: Sergey Streltsov

### PR DESCRIPTION
**Pull request by Mantikor**

### CODE REVIEW

File: src/years.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+1992: ##
+1963: ##
+1955: #
+1925: #
+1921: #
+1926: #
+2015: #
+1965: #
+1942: #
+2013/I: #
```
Comment (belovmd) [inline src/years.txt:71]: unclear year for me

File: src/years.txt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+1934: #
+1978: #
+1967: #
+1971: #
+1964: #
+1944: #
+1972: #
+1928: #
+2017/I: #
```
Comment (belovmd) [inline src/years.txt:80]: unclear year

File: src/years.txt [hunk 3]
```diff
@@ -0,0 +1,85 @@
+1995: #########
```
Comment (belovmd) [inline src/years.txt:1]: usually we sort column names not value

File: src/ratings.txt [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+8.2: #########################################
+8.3: #####################################
+8.0: ###################################
+8.5: #########################
+8.4: ###################
+8.6: ########
+8.7: #######
+8.9: ####
+8.8: ###
+9.2: ##
```
Comment (belovmd) [inline src/ratings.txt:11]: why is it unsorted?

File: src/imdb.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# coding: utf8
+
+"""
+EPAM python q4 homework 05 2
+Sergey Streltsov 2019-11-17
```
Comment (belovmd) [inline src/imdb.py:7]: missed task description

File: src/imdb.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+def load_ratings(filename):
+    run_path = os.path.dirname(os.path.abspath(__file__))
+    full_path = os.path.join(run_path, filename)
+    if os.path.isfile(full_path):
+        top250 = list()
+        is_top250 = False
+        is_header = False
+        with open(full_path, 'r', encoding='cp1252') as in_file:
+            for idx, line in enumerate(in_file):
```
Comment (belovmd) [inline src/imdb.py:23]: are you sure this enumerate wouldn't cause all file read? it looks to be risky from performance point of view

File: src/imdb.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                if 'TOP 250 MOVIES' in line:
+                    is_top250 = True
+                    continue
+                if is_top250 and ('Votes  Rank  Title' in line):
+                    is_header = True
+                    count = 0
+                    continue
+                if is_top250 and is_header:
+                    top250.append(line)
```
Comment (belovmd) [inline src/imdb.py:32]: overall it's good solution

File: src/runner.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    """Functions runner
+
+    Run functions from pytasks module, can be imported
+    or use as command line script
+    in command line you can specify function names
+    for call, like as runner.py fizzbuzz is_palindrome
+    """
+    if not args:
+        args = [func for func in dir(pytasks) if not func.startswith("__")]
+    [print(getattr(pytasks, func)()) for func in args]
```
Comment (belovmd) [inline src/runner.py]: it shouldn't print but just call

**Review by belovmd — COMMENTED** [review]
7,5

File: src/years.txt (hunk 1 above)
Comment (Mantikor) [reply src/years.txt:71]: 2013 year and I decade, this format provided by source file

File: src/imdb.py (hunk 6 above)
Comment (Mantikor) [reply src/imdb.py:23]: agree

File: src/years.txt (hunk 3 above)
Comment (Mantikor) [reply src/years.txt:1]: sorted by value for more readable

File: src/ratings.txt (hunk 4 above)
Comment (Mantikor) [reply src/ratings.txt:11]: sorted by value for more readable

File: src/imdb.py (hunk 5 above)
Comment (Mantikor) [reply src/imdb.py:7]: agree

**Pull request closed without merging**